projects/igniteui-angular/src/lib/services/transaction/igx-transaction.ts

```diff
@@ -166,12 +172,16 @@ export class IgxTransactionService<T extends Transaction, S extends State> exten
 
         const lastActions: Action<T>[] = this._undoStack.pop();
         this._transactions.splice(this._transactions.length - lastActions.length);
+        this._validationTransactions.splice(this._validationTransactions.length - lastActions.length);
         this._redoStack.push(lastActions);
 
         this._states.clear();
+        this._validationStates.clear();
         for (const currentActions of this._undoStack) {
             for (const transaction of currentActions) {
                 this.updateState(this._states, transaction.transaction, transaction.recordRef);
+                const validationTransaction = this._validationTransactions.find(x => x.id === transaction.transaction.id);
```

Comment:
Right, so I'm of the opinion that this form of state tracking for validity will likely slip at one point or another - unlike the value states being tracked that should aggregate neatly, validity doesn't. With more involved custom validation, where for example you have checks that depend on multiple cell values and the ability to commit transactions out of order per record id, grabbing the previous validity on undo in not guaranteed to be correct. That of course, is true for cell edits, but the same case can be made for cross-row checks even if less likely. If the validation depends on some external factor - then we have no idea if the old state is still valid regardless. 

I think the most prudent way to avoid having to deal with this is simply to keep a singular current state of validity and simply re-trigger validation on such state changes to update it - there aren't that many and it seems about in line with how most validation I've seen handles changes anyway.
Given that, you'll no longer need to store separate validation transactions, and with them gone and the state stored already being pretty foreign to the transaction service I don't see a reason to keep it there. Transactions have no control over the validation process and them being the owner of a state they makes even less sense to me this way, so we might as well move it out.